---
title: "Phylo tree methods"
author: "Sophia Horigan"
date: "6/30/2022"
output:
  pdf_document: default
  html_document: default
---

## Building a phylogenetic tree

This tutorial outlines methods for building a maximum-likelihood phylogenetic tree using the program RAxML, and time/phylogeography trees using BEAST1 and BEAST2

############################################
### Full-genome Astrovirus phylogeny
############################################

1. I scraped full genome non-bat host **reference** nucleotide sequences from  NCBI-virus (https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/).
Search:
* Astroviridae (tax id 39733)
* unclassified Astroviridae (tax id 352926)

with selections Sequence Type = RefSeq, RefSeq Genome Completeness = complete.
Total = 49

2. I scraped full genome bat nucleotide sequences from NCBI-virus (https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/).
Search:
* Astroviridae (tax id 39733)
* unclassified Astroviridae (tax id 352926)

with selections Nucleotide Completeness = complete, Host = Chiroptera (tax id 30560)
Total = 10

49 + 10 + 2 (my full) = 61 sequences
 
3. Cutoffs
To help alignment, I decided to cut any sequences that were less than 6kb (based on length)
This left 48 sequences.

3. Outgroup
Inlcuding one turkey avastrovirus as the outgroup. thus, I removed all other avastrovirus sequences. so now I have 43 total sequences.

4. Alignment
I aligned the 43 complete sequences in Genious using Multiple Alignment -> MAFFT on default settings.

5. I annotated areas with 50% or more gaps by using the Mask alignment function under the Tools menu.This will allow me to optionally remove the gaps when making a phylogeny. not going to do that with the first tree.

5. I used PhyML online server to estimate the nucleotide substitution model
http://www.atgc-montpellier.fr/phyml/
GTR is the default (gamma or gamma+I are the options) is R the same as gamma? am I dumb?

RAxML tree (not RAxML-NG tree like Cara did--RAxML also used for taxon rich datasets primarily: https://academic.oup.com/bioinformatics/article/35/21/4453/5487384)
-have to select the number of bootstraps (duh)
-default is FPB (felenstein boostraps) vs TBE. according to this paper, TBE is only introduced when using a large dataset with thousands of taxa: https://www.biorxiv.org/content/10.1101/154542v1.full.pdf

500 bootstrap on my laptop took ~48 hours. Need to install RAxML on the cluster.

6. moved consensus tree file to R studio to make the figure.



############################################
### ML RdRp tree - Madagascar 
############################################

Create an ML RdRp tree using all published Madagascar seqs and our new Mada seqs

CREATE DATASET
1. determine how many of my samples have the complete RdRp gene
-we only have 2, contained within the full genomes

2. get all published Madabat RdRp seqs
-2 sources:
  -1: Hoarau et al 2021 : https://virologyj.biomedcentral.com/articles/10.1186/s12985-021-01673-2
  -2: Lebarbenchon et al 2017 : https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5520320/
  
  total seqs (with my 2 novel = 67)

ALIGNMENT
3. Align in Geneious prime using MAFFT default settings and trim alignment

4. Export alignment as .fasta to modeltest-ng folder

MODELTEST-NG
5. scp folder with alignment and batch script to midway
`scp -r /Users/sophiahorigan/Documents/GitHub/Mada-Bat-Astro/modeltest-ng/ shorigan@midway2.rcc.uchicago.edu:/home/shorigan/`

6. execute batch script
`chmod 0700 modeltestbash.sbatch`
`./modeltestbash.sbatch`
##
contents of modeltestbash.sbach:
#!/bin/bash
#SBATCH --job-name=batAstro
#SBATCH --account=cbrook
#SBATCH --partition=broadwl
#SBATCH --nodes=1
#SBATCH --ntasks-per-node=8
#SBATCH --cpus-per-task=1
#SBATCH --time=36:00:00

module load python

conda activate /project2/cbrook/software/modeltest_env

modeltest-ng -i alignment.fasta -t ml -p 8

scp -r shorigan@midway2.rcc.uchicago.edu:/home/shorigan/modeltest-ng/alignment.fasta.out /Users/sophiahorigan/Documents/GitHub/Mada-Bat-Astro/modeltest-ng/

conda deactivate
##

output : 
Summary:

Partition 1/1:
                         Model         Score        Weight
----------------------------------------------------------
       BIC         TPM2uf+I+G4    28051.7731        0.5345
       AIC            TVM+I+G4    27482.2826        0.7046
      AICc            TVM+I+G4    27617.2826        0.7088


RAXML-NG
6. Export alignment as .phy to raxml-ng directory, and move directory to midway

7. load raxml-ng
`module load openmpi/3.0.0+gcc-10.1.0`
`export PATH=/project2/cbrook/software/raxml-ng/bin:$PATH`

8. make sure it will work with your specific model and alignment
`raxml-ng-mpi --check --msa alignment.phy --model TPM2uf+I+G4 --prefix T1`

9. check for number of recommended threads
`raxml-ng-mpi --parse --msa alignment.phy --model TPM2uf+I+G4 --prefix T2`
in this case, recommended threads/MPI processes = 1

10. modify sbatch script to include correct model and number of threads

in this case, looks like this...
#!/bin/bash
#SBATCH --job-name=raxml-ng
#SBATCH --account=cbrook
#SBATCH --partition=broadwl
#SBATCH --nodes=1
#SBATCH --ntasks-per-node=8
#SBATCH --cpus-per-task=1
#SBATCH --time=36:00:00
#SBATCH --mail-type=ALL
#SBATCH --mail-user=shorigan@uchicago.edu


`raxml-ng-mpi --all --msa alignment.phy --model TPM2uf+I+G4 --prefix T3 --seed 2 --threads 1 --bs-metric fbp,tbe --bs-trees 1000`

Update: had to manually set the number of bootstraps at 1000, and it did not converge! but was close. going to add 200 more bootstraps 

Check convergence using 
`raxml-ng-mpi --bsconverge --bs-trees T3.raxml.bootstraps --prefix T4 --seed2 2 --threads 1`

Run 200 more bootstraps
`raxml-ng --bootstrap --msa alignment.phy --model TPM2uf+I+G4 --prefix T5 --seed 333 --threads 1 --bs-trees 200 --bs-metric fbp,tbe`

Concatenate bootstraps
`cat T3.raxml.bootstraps T5.raxml.bootstraps > allbootstraps`

Re-assess convergence
`raxml-ng-mpi --bsconverge --bs-trees allbootstraps --prefix T6 --seed 2 --threads 1 --bs-cutoff 0.03`

Not converged yet! but still getting there. going to add 500 more bootstraps (T7 - T2)
`raxml-ng --bootstrap --msa alignment.phy --model TPM2uf+I+G4 --prefix T7 --seed 666 --bs-trees 500 --bs-metric fbp,tbe`

Hooray, converged after 1650 bootstraps!

Map BS support values onto best-scoring/best-known ML tree, using .besttree generated in the original run
`raxml-ng-mpi --support --tree T3.raxml.bestTree --bs-trees allbootstraps --prefix T13 --threads 1`

Tree with support saved as
`T3.raxml.supportFPB` and `T3.raxml.supportTBE`

Scp back to local computer, import into R to use with phylogenies.R to make trees
1. copy contents into new directory and save on Github (so I can delete and re-upload new empty version of raxml-ng)
`cp -r raxml-ng/* raxml-ng-ML-RdRp-Mada`


############################################
### ML RdRp tree - SWIO 
############################################
Goal: make an ML tree including RdRp seqs from SWIO area to start investigating some biogeographical patterns 

GATHER SEQUENCES
1. To the file containing the Madagascar RdRp seqs from the previous tree (plus our two novel seqs), I added SWIO seqs that I found as a result of a lit search. I located seqs from Reunion Island and Mozambique. Altogether, there are now 125 sequences.

ALIGNMENT
1. In Geneious Prime, I aligned the RdRp sequences using default settings. (same turkey avastrovirus outgroup as other trees)

2. Didn't even need to trim becuase all of the sequences were basically the same length, woohoo!

MODEL TEST
1. export alignment as a .fasta for model test

2. put into modeltest-ng directory

3. move directory to midway and run

output: 
AIC	  model              K            lnL          score          delta    $
--------------------------------------------------------------------------------
       1  GTR+I+G4          10     -7130.6432     14431.2864         0.0000    $
       2  TIM2+I+G4          8     -7134.0639     14434.1277         2.8413    $
       3  TVM+I+G4           9     -7133.2481     14434.4961         3.2097    $
       4  TIM3+I+G4          8     -7134.7231     14435.4462         4.1599    $
       5  TrN+I+G4           7     -7137.0093     14438.0185         6.7322    $
       6  TIM1+I+G4          8     -7136.0322     14438.0644         6.7780    $
       7  TPM2uf+I+G4        7     -7137.1020     14438.2040         6.9177    $
       8  TIM2ef+I+G4        5     -7139.6317     14439.2633         7.9770    $
       9  TPM3uf+I+G4        7     -7137.8550     14439.7099         8.4235    $
      10  SYM+I+G4           7     -7138.6497     14441.2995        10.0131    $
--------------------------------------------------------------------------------

GTR+I+G4 is the winner!
















BAYESIAN TIMETREE
Going to make a tree using BEAST to infer MRCA of group of SWIO seqs. 
*made a directory in the Github folder called beast

Generate .xml for Bayesian timetree using the program BEAUti (Bayesian Evolutionary Analysis Utility)


slurm script
#!/bin/bash
#SBATCH --job-name=beast
#SBATCH --account=pi-cbrook
#SBATCH --partition=broadwl
#SBATCH --nodes=1
#SBATCH --ntasks-per-node=8
#SBATCH --ntasks=8
#SBATCH --time=36:00:00

module load midway2
module load java/1.8
module load cuda/8.0
module load beagle/trunk
module load beast/2.6.2

beast -beagle_CPU -seed 777 /home/shorigan/beast/rdrp_swio.xml





