---
title: "Phylo tree methods"
author: "Sophia Horigan"
date: "6/30/2022"
output:
  pdf_document: default
  html_document: default
---

## Building a phylogenetic tree

This tutorial outlines methods for building a maximum-likelihood phylogenetic tree using the program RAxML.

### Full-genome Astrovirus phylogeny

1. I scraped full genome non-bat host **reference** nucleotide sequences from  NCBI-virus (https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/).
Search:
* Astroviridae (tax id 39733)
* unclassified Astroviridae (tax id 352926)

with selections Sequence Type = RefSeq, RefSeq Genome Completeness = complete.
Total = 49

2. I scraped full genome bat nucleotide sequences from NCBI-virus (https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/).
Search:
* Astroviridae (tax id 39733)
* unclassified Astroviridae (tax id 352926)

with selections Nucleotide Completeness = complete, Host = Chiroptera (tax id 30560)
Total = 10

49 + 10 + 2 (my full) = 61 sequences
 
3. Cutoffs
To help alignment, I decided to cut any sequences that were less than 6kb (based on length)
This left 48 sequences.

3. Outgroup
Inlcuding one turkey avastrovirus as the outgroup. thus, I removed all other avastrovirus sequences. so now I have 43 total sequences.

4. Alignment
I aligned the 43 complete sequences in Genious using Multiple Alignment -> MAFFT on default settings.

5. I annotated areas with 50% or more gaps by using the Mask alignment function under the Tools menu.This will allow me to optionally remove the gaps when making a phylogeny. not going to do that with the first tree.

5. I used PhyML online server to estimate the nucleotide substitution model
http://www.atgc-montpellier.fr/phyml/
GTR is the default (gamma or gamma+I are the options) is R the same as gamma? am I dumb?

RAxML tree (not RAxML-NG tree like Cara did--RAxML also used for taxon rich datasets primarily: https://academic.oup.com/bioinformatics/article/35/21/4453/5487384)
-have to select the number of bootstraps (duh)
-default is FPB (felenstein boostraps) vs TBE. according to this paper, TBE is only introduced when using a large dataset with thousands of taxa: https://www.biorxiv.org/content/10.1101/154542v1.full.pdf

500 bootstrap on my laptop took ~48 hours. Need to install RAxML on the cluster.

6. moved consensus tree file to R studio to make the figure.



############################################
### Partial genome RdRp from positive samples
############################################

First attempt at a gene tree. The gene I idendified is RdRp (rna something polymerase protein). it is apparently highly conserved, and used in many phylogenetic analyses.

1. determine how many of my samples have the complete RdRp gene
- turns out it's just 2, the full genomes. all other hits are pretty small fragments, much smaller than the 1.5k bp RdRp, and in an alignment don't match up with any part of the RdRp from the full genomes.
(PS: I annotated the full genomes from a reference mamastrovirus downloaded from within Geneious from NCBI virus)

2. For this first tree, am using all of the same full genome seqs I used for the full genome tree (because I know they have RdRp by nature of being full). So I used Geneious to import the annotated version of all of them, went through and cleaned up the annotation, labeling ORF1b as 'RdRp', then extracted the RdRp gene from each genome and put them into a new folder called 'RdRp_tree'

3. after this, I made an alignment using MAFFT default settings, and decided to delete the 2 bastrovirus seqs (what is that? online says some sort of henepavirus/astrovirus hybrid..) because it was so divergent. *may need/want to do this for the full-genome phylogeny also.

4. Realigned. Visually trimmed the ends a bit.

5. Exported alignment to PhyML online to determine nucleotide substition model (modeltest-ng still being installed on the cloud)

6. Made RAxML tree in Geneious. Prelim tree for poster only 50 bootstraps (hopefully this finishes by tomorrow morning)
















