---
title: "Phylo tree methods"
author: "Sophia Horigan"
date: "6/30/2022"
output:
  pdf_document: default
  html_document: default
---

## Building a phylogenetic tree

This tutorial outlines methods for building a maximum-likelihood phylogenetic tree using the program RAxML.

### Full-genome Astrovirus phylogeny

1. I scraped full genome non-bat host **reference** nucleotide sequences from  NCBI-virus (https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/).
Search:
* Astroviridae (tax id 39733)
* unclassified Astroviridae (tax id 352926)

with selections Sequence Type = RefSeq, RefSeq Genome Completeness = complete.
Total = 49

2. I scraped full genome bat nucleotide sequences from NCBI-virus (https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/).
Search:
* Astroviridae (tax id 39733)
* unclassified Astroviridae (tax id 352926)

with selections Nucleotide Completeness = complete, Host = Chiroptera (tax id 30560)
Total = 10

49 + 10 + 2 (my full) = 61 sequences
 
3. Cutoffs
To help alignment, I decided to cut any sequences that were less than 6kb (based on length)
This left 48 sequences.

3. Outgroup
Inlcuding one turkey avastrovirus as the outgroup. thus, I removed all other avastrovirus sequences. so now I have 43 total sequences.

4. Alignment
I aligned the 43 complete sequences in Genious using Multiple Alignment -> MAFFT on default settings.

5. I annotated areas with 50% or more gaps by using the Mask alignment function under the Tools menu.This will allow me to optionally remove the gaps when making a phylogeny. not going to do that with the first tree.

5. I used PhyML online server to estimate the nucleotide substitution model
http://www.atgc-montpellier.fr/phyml/
GTR is the default (gamma or gamma+I are the options) is R the same as gamma? am I dumb?

RAxML tree (not RAxML-NG tree like Cara did--RAxML also used for taxon rich datasets primarily: https://academic.oup.com/bioinformatics/article/35/21/4453/5487384)
-have to select the number of bootstraps (duh)
-default is FPB (felenstein boostraps) vs TBE. according to this paper, TBE is only introduced when using a large dataset with thousands of taxa: https://www.biorxiv.org/content/10.1101/154542v1.full.pdf

500 bootstrap on my laptop took ~48 hours. Need to install RAxML on the cluster.

6. moved consensus tree file to R studio to make the figure.



############################################
### ML RdRp tree - Madagascar 
############################################

Create an ML RdRp tree using all published Madagascar seqs and our new Mada seqs

CREATE DATASET
1. determine how many of my samples have the complete RdRp gene
-we only have 2, contained within the full genomes

2. get all published Madabat RdRp seqs
-2 sources:
  -1: Hoarau et al 2021 : https://virologyj.biomedcentral.com/articles/10.1186/s12985-021-01673-2
  -2: Lebarbenchon et al 2017 : https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5520320/
  
  total seqs (with my 2 novel = 67)

ALIGNMENT
3. Align in Geneious prime using MAFFT default settings and trim alignment

4. Export alignment as .fasta to modeltest-ng folder

MODELTEST-NG
5. scp folder with alignment and batch script to midway
`scp -r /Users/sophiahorigan/Documents/GitHub/Mada-Bat-Astro/modeltest-ng/ shorigan@midway2.rcc.uchicago.edu:/home/shorigan/`

6. execute batch script
`chmod 0700 modeltestbash.sbatch`
`./modeltestbash.sbatch`
##
contents of modeltestbash.sbach:
#!/bin/bash
#SBATCH --job-name=batAstro
#SBATCH --account=cbrook
#SBATCH --partition=broadwl
#SBATCH --nodes=1
#SBATCH --ntasks-per-node=8
#SBATCH --cpus-per-task=1
#SBATCH --time=36:00:00

module load python

conda activate /project2/cbrook/software/modeltest_env

modeltest-ng -i alignment.fasta -t ml -p 8

scp -r shorigan@midway2.rcc.uchicago.edu:/home/shorigan/modeltest-ng/alignment.fasta.out /Users/sophiahorigan/Documents/GitHub/Mada-Bat-Astro/modeltest-ng/

conda deactivate
##

output : 
Summary:

Partition 1/1:
                         Model         Score        Weight
----------------------------------------------------------
       BIC         TPM2uf+I+G4    28051.7731        0.5345
       AIC            TVM+I+G4    27482.2826        0.7046
      AICc            TVM+I+G4    27617.2826        0.7088


RAXML-NG
6. Export alignment as .phy to raxml-ng directory, and move directory to midway

7. load raxml-ng
`module load openmpi/3.0.0+gcc-10.1.0`
`export PATH=/project2/cbrook/software/raxml-ng/bin:$PATH`

8. make sure it will work with your specific model and alignment
`raxml-ng-mpi --check --msa alignment.phy --model TPM2uf+I+G4 --prefix T1`

9. check for number of recommended threads
`raxml-ng-mpi --parse --msa alignment.phy --model TPM2uf+I+G4 --prefix T2`
in this case, recommended threads/MPI processes = 1

10. modify sbatch script to include correct model and number of threads

in this case, looks like this...
#!/bin/bash
#SBATCH --job-name=raxml-ng
#SBATCH --account=cbrook
#SBATCH --partition=broadwl
#SBATCH --nodes=1
#SBATCH --ntasks-per-node=8
#SBATCH --cpus-per-task=1
#SBATCH --time=36:00:00
#SBATCH --mail-type=ALL
#SBATCH --mail-user=shorigan@uchicago.edu


raxml-ng-mpi --all --msa alignment.phy --model TPM2uf+I+G4 --prefix T3 --seed 2 --threads 1 --bs-metric fbp,tbe

11. change permissions and execute batch script






