---
title: "Phylo tree methods"
author: "Sophia Horigan"
date: "6/30/2022"
output:
  pdf_document: default
  html_document: default
---

## Building a phylogenetic tree

This tutorial outlines methods for building a maximum-likelihood phylogenetic tree using the program RAxML.

### Full-genome Astrovirus phylogeny

1. I scraped full genome non-bat host **reference** nucleotide sequences from  NCBI-virus (https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/).
Search:
* Astroviridae (tax id 39733)
* unclassified Astroviridae (tax id 352926)

with selections Sequence Type = RefSeq, RefSeq Genome Completeness = complete.
Total = 49

2. I scraped full genome bat nucleotide sequences from NCBI-virus (https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/).
Search:
* Astroviridae (tax id 39733)
* unclassified Astroviridae (tax id 352926)

with selections Nucleotide Completeness = complete, Host = Chiroptera (tax id 30560)
Total = 10

49 + 10 + 2 (my full) = 61 sequences
 
3. Cutoffs
To help alignment, I decided to cut any sequences that were less than 6kb (based on length)
This left 48 sequences.

3. Outgroup
I am including a few avastrovirus sequences in the tree to use as an outgroup. another paper does this for a gene tree, but I think it makes sense for a full genome tree as well. Might only need one though (other paper uses a few though)

4. Alignment
I aligned the 48 complete sequences in Genious using Multiple Alignment -> MAFFT on default settings.

5. I annotated areas with 50% or more gaps by using the Mask alignment function under the Tools menu.This will allow me to optionally remove the gaps when making a phylogeny. not going to do that with the first tree.

5. I used PhyML online server to estimate the nucleotide substitution model
http://www.atgc-montpellier.fr/phyml/
Bayesian criterion model: GTR +R
AIC model: 


### Partial genome RdRp from positive samples

To make Fig3b, I did the following:
1. Downloaded all the non-host reads off of IDseq from all 30 positive bats in our dataset (actually all 27 for which we did not already have full genome).
2. Re-uploaded these into IDseq, checking the appropriate species as the host.
3. Clicked on the "consensus genome" option after upload
2. Loaded into Geneious and did a multi-sequence alignment to get an idea of what we were looking at.
